cpp/src/arrow/buffer.h

```diff
@@ -43,7 +43,7 @@ class Status;
 /// of bytes that where allocated for the buffer in total.
 ///
 /// The following invariant is always true: Size < Capacity
-class ARROW_EXPORT Buffer : public std::enable_shared_from_this<Buffer> {
```

Comment:
This is a bit of a red herring for this patch, but I ran across this cruft and decided to remove it. I can back out this change if there's concern